### PR TITLE
Add ObjectProcessor noop impl and ObjectProcessor tests

### PR DESCRIPTION
**Pull request by devinrsmith**
Unit test follow-up to #4648

### CODE REVIEW

File: extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java [hunk 1]
```diff
@@ -66,6 +66,22 @@ static <T> ObjectProcessor<T> rowLimited(ObjectProcessor<T> delegate, int rowLim
...
 
+    /**
+     * Creates a "no-operation" object processor that ignores the input object chunk. If {@code fillWithNullValue} is
+     * {@code true}, during {@link ObjectProcessor#processAll(ObjectChunk, List) processAll}
+     * {@link WritableChunk#fillWithNullValue(int, int) fillWithNullValue} will be invoked on each output chunk;
+     * otherwise, the output chunk contents will not be modified. In either case, the processing will increment the
+     * output chunks sizes.
+     *
+     * @param outputTypes the output types
+     * @param fillWithNullValue if the output chunks should be filled with the appropriate null value;
```
Comment (rcaudy) [inline extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java:77]:
```suggestion
     * @param fillWithNullValue if the output chunks should be filled with the appropriate null value
```

File: extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java [hunk 2]
```diff
@@ -66,6 +66,22 @@ static <T> ObjectProcessor<T> rowLimited(ObjectProcessor<T> delegate, int rowLim
...
+     * @return the no-op object processor
+     * @param <T> the object type
+     */
+    static <T> ObjectProcessor<T> noop(List<Type<?>> outputTypes, boolean fillWithNullValue) {
```
Comment (rcaudy) [inline extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java:81]: Does this belong in production sources? Same question for `ObjectProcessorNoop`, of course.

File: extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessorStrict.java [hunk 3]
```diff
@@ -22,22 +22,28 @@ static <T> ObjectProcessor<T> create(ObjectProcessor<T> delegate) {
...
+        this.outputTypes = List.copyOf(delegate.outputTypes());
     }
 
     @Override
     public List<Type<?>> outputTypes() {
-        return delegate.outputTypes();
+        final List<Type<?>> outputTypes = delegate.outputTypes();
+        if (!this.outputTypes.equals(outputTypes)) {
+            throw new UncheckedDeephavenException("Implementation is returning a different list of outputTypes");
+        }
```
Comment (rcaudy) [inline extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessorStrict.java:37]: Do you think this is actually a real worry?

File: extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java (hunk 2 above)
Comment (devinrsmith) [reply extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java:81]: I think yes - it's potentially useful scaffolding to test out a given publisher before you have your own ObjectProcessor impl.

For example, let's say I want to develop my own Capn Proto ObjectProcessor, but in parallel I still want to allow other developers to work on the downstream queries while the impl is still in progress - they can become "unblocked" by faking it w/ noop:

```groovy
my_table = consumeToTable(..., objectProcessorSpec(noop([Type.stringType(), Type.intType()], true), ["Name", "Age"]))

File: extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessorStrict.java (hunk 3 above)
Comment (devinrsmith) [reply extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessorStrict.java:37]: No, but the strict impl _is_ being a bit pedantic on purpose.

**Review by rcaudy — COMMENTED** [review]
Placeholder

File: extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java (hunk 2 above)
Comment (rcaudy) [reply extensions/kafka/src/main/java/io/deephaven/processor/ObjectProcessor.java:81]: That seems pretty tenuous.

**Review by rcaudy — APPROVED** [review]

**Pull request merged**